packages/punchcard/lib/appsync/vtl-object.ts

```diff
@@ -0,0 +1,208 @@
+import { AnyShape, BinaryShape, bool, integer, NeverShape, NothingShape, ShapeGuards, ShapeVisitor, UnknownShape, Value } from '@punchcard/shape';
+import { ArrayShape, BoolShape, DynamicShape, IntegerShape, MapShape, NumberShape, Pointer, RecordShape, SetShape, Shape, StringShape, TimestampShape } from '@punchcard/shape';
+import { string, Trait } from '@punchcard/shape';
+import { FunctionArgs, FunctionShape } from '@punchcard/shape/lib/function';
+import { VExpression } from './expression';
+
+const type = Symbol.for('GraphQL.Type');
+const expr = Symbol.for('GraphQL.Expression');
+
+export class VObject<T extends Shape = Shape> {
```

Comment:
Why is the fallback type needed here?... won't it be inferred as the constructor's 1st arg?... or would removing it result in a 1-generic-expected TS error?